packages/cadl-ranch-specs/http/resiliency/srv-driven/main.tsp

```diff
@@ -0,0 +1,196 @@
+  3. Api version: The initial deployment of the service only supports api version 'v1'. The new deployment of the service supports api versions 'v1' and 'v2'.
+
+  With the above two calls, we test the following configurations from this service spec:
+  - A client generated from the second service spec can call the second deployment of a service with api version v1
+  - A client generated from the second service spec can call the second deployment of a service with api version v2 with the updated changes
+
+  Tests that we can grow up an operation from accepting one optional parameter to accepting two optional parameters.
+  """)
+  @route("/from-one-optional")
+  @doc("Tests that we can grow up an operation from accepting one optional parameter to accepting two optional parameters.")
+  @get
+  fromOneOptional(
+    @doc("I am an optional parameter")
+    @query
+    parameter?: string,
+
+    @added(Versions.v2)
+    @doc("I'm a new input optional parameter")
+    @query
+    `new-parameter`?: string
+  ): NoContentResponse;
+}
+
+@scenario
+@scenarioDoc("""
+  Need the following two calls:
+  - Pass in `serviceDeploymentVersion="v2"` and `apiVersion="v1"` with header `content-type="application/json"` and expected request body:
+  ```json
+  { url: \"http://example.org/myimage.jpeg\" }
+  ```
+  - Pass in `serviceDeploymentVersion="v2"` and `apiVersion="v2"` with header `content-type="image/jpeg"` and expected request body:
+  ```json
+  { url: \"http://example.org/myimage.jpeg\" }
+  ```
+
+  There are three concepts that should be clarified:
+  1. Client spec version: refers to the spec that the client is generated from. 'v1' is a client generated from old.tsp and 'v2' is a client generated from main.tsp.
+  2. Service deployment version: refers to a deployment version of the service. 'v1' represents the initial deployment of the service with a single api version. 'v2' represents the new deployment of a service with multiple api versions
+  3. Api version: The initial deployment of the service only supports api version 'v1'. The new deployment of the service supports api versions 'v1' and 'v2'.
+
+  With the above two calls, we test the following configurations from this service spec:
+  - A client generated from the second service spec can call the second deployment of a service with api version v1
+  - A client generated from the second service spec can call the second deployment of a service with api version v2 with the updated changes
+
+  Tests that we can grow up an operation from accepting one content type to two.
+  """)
+@route("/add-content-type")
+@doc("Operation that we can grow up an operation from accepting one content type to two.")
+@post
+op addContentType(
+  @header `content-type`: "application/json" | "image/jpeg",
+
+  @doc("I am a body parameter. My only valid JSON entry is { url: \"http://example.org/myimage.jpeg\" }")
+  @body
+  parameter: PostInput
+): NoContentResponse;
```

Comment:
I seems not able to find any typespec sematic saying that `"image/jpeg"` is added in v2?

I am not against the case, but feeling this may be not mature enough. (i.e. maybe sharedRoute can handle this better, via adding another operation).

Reply:
yeah you are right, maybe we should omit this test until we have a better understanding.